include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -472,4 +472,24 @@ def RefImpl : FIRRTLImplType<"Ref",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Non-Hardware Type Definitions
+//===----------------------------------------------------------------------===//
+
+def StringImpl : FIRRTLImplType<
+  "String", [PhasedTypeInterface], "circt::firrtl::FIRRTLType"> {
+  let summary = [{
+    an unlimited length string type. Not representable in hardware.
+    Commonly used as omir properties.
+  }];
+  let parameters = (ins "Phase":$phase);
```

Comment:
Just curios, why was it necessary to create a parameter `phase` even when `phase` must be always `property`?

Reply:
I set it up this way so that if/when we have parameters, strings can also be in the parameter phase.